File: Array/leftRotation.py
# Left Rotation of array

import sys

def leftRotation(a, d):
    finalArray = [[] for _ in range(lenArray)]
    for i in range(len(a)):
        index = i - d
        if index < 0:
            index = len(a) + index
        finalArray[index]= a[i]
    return finalArray
        

if __name__ == "__main__":
    n, d = raw_input().strip().split(' ')
    n, d = [int(n), int(d)]
    a = map(int, raw_input().strip().split(' '))
    finalArray = leftRotation(a, d)
    print " ".join(map(str, finalArray))

File: Array/leftRotation.js
// Left Rotation of array

process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

//console.log('Ctrl + C to get the result===>')

process.stdin.on('data', function(data) {
    input_stdin += data;
});

process.on('SIGINT', function() {
    input_stdin_array = input_stdin.split("\n");
    main();
    process.exit();
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

/////////////// ignore above this line ////////////////////

function main() {
    d = parseInt(readLine());
    a = readLine().split(' ');
    finalArray = new Array(lenArray);
    for (i = 0; i < a.length; i++) {
        var index = i - d;
        if (index < 0) {
            index = a.length + index;
        }
        finalArray[index] = a[i];
    }
    console.log(finalArra);
}